### PR TITLE
exchanges/Bybit: Fix UpdateOrderExecutionLimits

### PR DESCRIPTION
**Pull request by gloriousCode**
# PR Description

Addresses Issue https://github.com/thrasher-corp/gocryptotrader/issues/1729
### Issue replication steps:
1. Enable Bybit
2. Enable USDCMarginedFutures asset and pairs
3. Appreciate spam logs: `Bybit unable to load limits for POPCATUSDT, pair data missing`

### Fix details
We do some manipulation of currencypairs in `UpdateTradablePairs` for USDCMarginedFutures pairs. This update adds the formatting to `UpdateOrderExecutionLimits` so that pairs can match

#### Random extras
- The was a lint issue in assigning a value to uint8 when the original value is int64, I just changed it to int64. You can fight me on the memory improvements and how much slower GCT is now
- Had to update json data for options, which involved updating the options tradable pair
- Used NextPageCursor in more places

Fixes # (issue)
https://github.com/thrasher-corp/gocryptotrader/issues/1729
## Type of change
- [x] Bug fix (non-breaking change which fixes an issue)

## How has this been tested
- [x] go test ./... -race
- [x] golangci-lint run
- [x]  Update httprecording details

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation and regenerated documentation via the documentation tool
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally and on Github Actions with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by shazbert — CHANGES_REQUESTED** [review]
Looking good, fixes look great just some minor nits.

File: exchanges/bybit/bybit_test.go [hunk 1]
```diff
@@ -772,13 +769,13 @@ func TestUpdateOrderExecutionLimits(t *testing.T) {
...
-	enabled, err := b.GetAvailablePairs(asset.Spot)
+	availablePairs, err := b.GetAvailablePairs(asset.Spot)
 	if err != nil {
 		t.Fatal("Bybit GetAvailablePairs() error", err)
 	}
-	for x := range enabled {
+	for x := range availablePairs {
 		var limits order.MinMaxLevel
-		limits, err = b.GetOrderExecutionLimits(asset.Spot, enabled[x])
+		limits, err = b.GetOrderExecutionLimits(asset.Spot, availablePairs[x])
```
[the remark this answers is not in the document]
Comment (gloriousCode) [reply exchanges/bybit/bybit_test.go:778]: haha how did I miss that

File: exchanges/bybit/bybit_wrapper.go [hunk 2]
```diff
@@ -291,28 +291,35 @@ func (by *Bybit) FetchTradablePairs(ctx context.Context, a asset.Item) (currency
...
 		response *InstrumentsInfo
 	)
+	var nextPageCursor string
 	switch a {
 	case asset.Spot, asset.CoinMarginedFutures, asset.USDCMarginedFutures, asset.USDTMarginedFutures:
 		category = getCategoryName(a)
-		response, err = by.GetInstrumentInfo(ctx, category, "", "Trading", "", "", int64(by.Features.Enabled.Kline.GlobalResultLimit))
-		if err != nil {
-			return nil, err
+		for {
```
[the remark this answers is not in the document]
Comment (gloriousCode) [reply exchanges/bybit/bybit_wrapper.go:298]: I don't feel right putting in a fake nextPageCursor sorry. Is there anything you can reference me to?

**Comment by gloriousCode** [comment]
I sorta kinda changed how currencies were processed now for all assets. No longer is there 100_000AIDOGEUSDT but more sane 10000AIDOGE_USDT

It has increased the review threshold sorry

File: exchanges/bybit/bybit_wrapper.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (shazbert) [reply exchanges/bybit/bybit_wrapper.go:298]: It's what it gives back in the response and I used that and tested it with a restricted limit size. I didn't even check the docs. But either way you can resolve.

[the remark this answers is not in the document]
Comment (gloriousCode) [reply exchanges/bybit/bybit_wrapper.go:298]: Well it looked made up and docs don't say anything about that and their referenced `nextPageCursor` looks way different...
But your thing actually works and thank you for testing it 🌈  I'll change it

**Review by shazbert — CHANGES_REQUESTED** [review]
changes looks good but I am still getting a bunch of warnings can you please check them out?

```
Bybit unable to load limits for usdcmarginedfutures BTC-12JAN24, pair data missing
Bybit unable to load limits for usdcmarginedfutures BTC-19JAN24, pair data missing
Bybit unable to load limits for usdcmarginedfutures BTC-23FEB24, pair data missing
Bybit unable to load limits for usdcmarginedfutures BTC-26JAN24, pair data missing
Bybit unable to load limits for usdcmarginedfutures BTC-27SEP24, pair data missing
Bybit unable to load limits for usdcmarginedfutures BTC-28JUN24, pair data missing
Bybit unable to load limits for usdcmarginedfutures BTC-29MAR24, pair data missing
Bybit unable to load limits for usdcmarginedfutures ETH-12JAN24, pair data missing
Bybit unable to load limits for usdcmarginedfutures ETH-19JAN24, pair data missing
Bybit unable to load limits for usdcmarginedfutures ETH-23FEB24, pair data missing
Bybit unable to load limits for usdcmarginedfutures ETH-26JAN24, pair data missing
Bybit unable to load limits for usdcmarginedfutures ETH-27SEP24, pair data missing
Bybit unable to load limits for usdcmarginedfutures ETH-28JUN24, pair data missing
Bybit unable to load limits for usdcmarginedfutures ETH-29MAR24, pair data missing
Bybit unable to load limits for usdcmarginedfutures MATIC-PERP, pair data missing
Bybit unable to load limits for usdtmarginedfutures 1000000VINU-USDT, pair data missing
Bybit unable to load limits for usdtmarginedfutures 10000NFT-USDT, pair data missing
Bybit unable to load limits for usdtmarginedfutures 10000STARL-USDT, pair data missing
Bybit unable to load limits for usdtmarginedfutures AGIX-USDT, pair data missing
Bybit unable to load limits for usdtmarginedfutures ANT-USDT, pair data missing
Bybit unable to load limits for usdtmarginedfutures CEEK-USDT, pair data missing
Bybit unable to load limits for usdtmarginedfutures FET-USDT, pair data missing
Bybit unable to load limits for usdtmarginedfutures FITFI-USDT, pair data missing
Bybit unable to load limits for usdtmarginedfutures FRONT-USDT, pair data missing
Bybit unable to load limits for usdtmarginedfutures FUN-USDT, pair data missing
Bybit unable to load limits for usdtmarginedfutures GAL-USDT, pair data missing
Bybit unable to load limits for usdtmarginedfutures GPT-USDT, pair data missing
Bybit unable to load limits for usdtmarginedfutures KLAY-USDT, pair data missing
Bybit unable to load limits for usdtmarginedfutures LOOM-USDT, pair data missing
Bybit unable to load limits for usdtmarginedfutures MATIC-USDT, pair data missing
Bybit unable to load limits for usdtmarginedfutures MULTI-USDT, pair data missing
Bybit unable to load limits for usdtmarginedfutures OCEAN-USDT, pair data missing
Bybit unable to load limits for usdtmarginedfutures RNDR-USDT, pair data missing
Bybit unable to load limits for usdtmarginedfutures STRAX-USDT, pair data missing
Bybit unable to load limits for usdtmarginedfutures TOMI-USDT, pair data missing
Bybit unable to load limits for usdtmarginedfutures UNFI-USDT, pair data missing
Bybit unable to load limits for usdtmarginedfutures VGX-USDT, pair data missing
Bybit unable to load limits for usdtmarginedfutures WSM-USDT, pair data missing
Bybit unable to load limits for usdtmarginedfutures YFII-USDT, pair data missing
```

**Review by shazbert — APPROVED** [review]
Nevermind that was only in tests, running it with all pairs enabled doesn't have that issue.

**Review by thrasher- — APPROVED** [review]
tACK, thanks for resolving this!

**Pull request merged**